apps/rush-lib/src/utilities/AsyncRecycler.ts

```diff
@@ -77,6 +78,40 @@ export class AsyncRecycler {
     );
   }
 
+  /**
+   * This deletes all items under the specified folder, except for the items in the membersToExclude.
+   * To be conservative, a case-insensitive comparison is used for membersToExclude.
+   * The membersToExclude must be file/folder names that would match readdir() results.
+   */
+  public moveAllItemsInFolder(folderPath: string, membersToExclude?: ReadonlyArray<string>): void {
+    const resolvedFolderPath: string = path.resolve(folderPath);
+
+    const excludeSet: Set<string> = new Set<string>(
+      (membersToExclude || []).map(x => x.toUpperCase())
+    );
+
+    for (const memberName of fsx.readdirSync(resolvedFolderPath)) {
+      const memberPath: string = path.join(resolvedFolderPath, memberName);
+      const normalizedMemberName: string = memberName.toUpperCase();
+      if (!excludeSet.has(normalizedMemberName)) {
+        let shouldMove: boolean = false;
+        try {
+          const stats: fsx.Stats = fsx.lstatSync(memberPath);
+          shouldMove = stats.isDirectory();
+        } catch (error) {
+          // If we fail to access the item, assume it's not a folder
+        }
+        if (shouldMove) {
+console.log('==> Recycling: ' + memberPath);
```

Comment:
>console [](start = 0, length = 7)

What's with the weird indentation?

Reply:
That was a reminder to delete this debug code.  :-)

Reply:
That was a reminder to delete this debug code.  :-)

---
In reply to: [186247303](https://github.com/Microsoft/web-build-tools/pull/643#discussion_r186247303) [](ancestors = 186247303)